src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -456,6 +448,10 @@ private bool UpdateCacheSizeExceedsCapacity(CacheEntry entry, CoherentState cohe
             for (int i = 0; i < 100; i++)
             {
                 long newSize = sizeRead + entry.Size;
+                if (priorEntry != null && entry.Key == priorEntry.Key)
```

Comment:
> && entry.Key == priorEntry.Key

This is true by definition if `priorEntry` is not null. Should it be a `Debug.Assert` instead?